module/modmanager/manager.go

```diff
@@ -819,7 +827,11 @@ func (mgr *Manager) newOnUnexpectedExitHandler(mod *module) func(exitCode int) b
 				mgr.logger.Warnw("Error while re-adding resource to module",
 					"resource", name, "module", mod.cfg.Name, "error", err)
 				mgr.rMap.Delete(name)
-				delete(mod.resources, name)
+				func() {
+					mod.resourcesMu.Lock()
+					defer mod.resourcesMu.Unlock()
```

Comment:
I can appreciate the defensiveness, but in this case we should just unlock after the delete. rather than creating a function for the purpose of `defer`.